apps/ledger-live-mobile/src/newArch/features/Accounts/Navigator.tsx

```diff
@@ -0,0 +1,152 @@
+import { useSelector } from "react-redux";
+import { hasClosedNetworkBannerSelector } from "~/reducers/settings";
+import { urls } from "~/utils/urls";
+import SelectCrypto from "LLM/features/AssetSelection/screens/SelectCrypto";
+import SelectNetwork from "LLM/features/AssetSelection/screens/SelectNetwork";
+import SelectAccounts from "LLM/features/Accounts/screens/SelectAccounts";
+import SelectDevice, {
+  addAccountsSelectDeviceHeaderOptions,
+} from "LLM/features/DeviceSelection/screens/SelectDevice";
+import ConnectDevice, {
+  connectDeviceHeaderOptions,
+} from "LLM/features/DeviceSelection/screens/ConnectDevice";
+import StepHeader from "~/components/StepHeader";
+import { NavigationHeaderCloseButtonAdvanced } from "~/components/NavigationHeaderCloseButton";
+import { NavigationHeaderBackButton } from "~/components/NavigationHeaderBackButton";
+import ScanDeviceAccounts from "LLM/features/Accounts/screens/ScanDeviceAccounts";
+
+export default function Navigator() {
+  const { colors } = useTheme();
+  const { t } = useTranslation();
+  const route = useRoute();
+
+  const hasClosedNetworkBanner = useSelector(hasClosedNetworkBannerSelector);
+
+  const onClose = useCallback(() => {
+    track("button_clicked", {
+      button: "Close",
+      screen: route.name,
+    });
+  }, [route]);
+
+  const stackNavigationConfig = useMemo(
+    () => ({
+      ...getStackNavigatorConfig(colors, true),
+      headerRight: () => <NavigationHeaderCloseButtonAdvanced onClose={onClose} />,
+    }),
+    [colors, onClose],
+  );
+
+  const onConnectDeviceBack = useCallback((navigation: NavigationProp<Record<string, unknown>>) => {
+    track("button_clicked", {
+      button: "Back arrow",
+      page: ScreenName.ConnectDevice,
+    });
+    navigation.goBack();
+  }, []);
+
+  return (
+    <Stack.Navigator
+      screenOptions={{
+        ...stackNavigationConfig,
+        gestureEnabled: Platform.OS === "ios",
+      }}
+    >
+      {/* Select Crypto (see : apps/ledger-live-mobile/src/screens/AddAccounts/01-SelectCrypto.js) */}
+      <Stack.Screen
+        name={ScreenName.AddAccountsSelectCrypto}
+        component={SelectCrypto}
+        options={{
+          headerLeft: () => <NavigationHeaderBackButton />,
+          headerTitle: "",
+          headerRight: () => <NavigationHeaderCloseButtonAdvanced onClose={onClose} />,
+        }}
+      />
+
+      <Stack.Screen
+        name={ScreenName.SelectNetwork}
+        component={SelectNetwork}
+        options={{
+          headerLeft: () => <NavigationHeaderBackButton />,
+          headerTitle: "",
+          headerRight: () => (
+            <Flex alignItems="center" justifyContent="center" flexDirection="row">
+              {hasClosedNetworkBanner && (
+                <HelpButton eventButton="Choose a network article" url={urls.chooseNetwork} />
+              )}
+              <NavigationHeaderCloseButtonAdvanced onClose={onClose} />
+            </Flex>
+          ),
+        }}
+      />
```

Comment:
Don't we want to add the following `Stack.Screen` definitions in a dedicated Navigator under `features/AssetSelection`?

Reply:
i did it locally, my problem now is how to use these nested navigator and compound them together, knowing that the entry point is "AddAccount" for buttons and deeplinks. Other challenge that i'm facing is that each navigator has header/back/previous control configuration so the first nesting give me this 
<img width="286" alt="image" src="https://github.com/user-attachments/assets/4401a312-fa71-49d1-adaf-141d8924f839">

Working on it , i totally agree about the concept, but hope it don't complicate things further.

Reply:
I've used an existing way to do so, `navigate(NavigatorName, {screen: ScreenName, params:...})`

Reply:
Your solution seems very goood too , while marketnavigator is used as a Stack group no ?